app/api/resources/resolver_view.py

```diff
@@ -14,36 +14,80 @@
 router = APIRouter()
 @router.get("/{stable_id}", name="Resolver")
 def resolve(request: Request, stable_id: str, type: Optional[str] = "gene", gca: Optional[str] = "", app: Optional[str] = DEFAULT_APP):
+
   params = SearchPayload(
     stable_id = stable_id,
-    type = type
+    type = type,
+    per_page = 10
   )
+
+  # Get genome_ids from search api
   try:
     session = requests.Session()
     search_results = {}
     with session.post(url=ENSEMBL_SEARCH_HUB_API,json=params.dict()) as response:
       response.raise_for_status()
       search_results = response.json()
-  
   except requests.exceptions.HTTPError as HTTPError:
     return response_error_handler({"status": HTTPError.response.status_code})
 
   except Exception as e:
     logger.exception(e)
     return response_error_handler({"status": 500})
-
-  if request.headers.get("content-type") == "application/json":
-    return search_results
   else:
     matches = search_results.get("matches")
-    if not matches:
-      return {"response": "No matches found"}
 
-    genome_id = matches[0].get("genome_id")
+  if not matches:
+    return response_error_handler({"status": 404})
+
+  # Get assembly information
+  meta_session = requests.Session()
+  meta_results = {}
+  resolved_genome_info = []
 
-    if (app == "entity-viewer"):
-      url=f"{ENSEMBL_URL}/{app}/{genome_id}/{type}:{stable_id}"
+  for match in matches:
+    genome_id = match.get("genome_id")
+    try:
+      with meta_session.get(url=f"{ENSEMBL_URL}/api/metadata/genome/{genome_id}/details") as meta_response:
+        meta_response.raise_for_status()
+    except requests.exceptions.HTTPError as HTTPError:
+      return response_error_handler({"status": HTTPError.response.status_code})
+    except Exception as e:
+      logger.exception(e)
+      return response_error_handler({"status": 500})
     else:
-      url=f"{ENSEMBL_URL}/{app}/{genome_id}?focus={type}:{stable_id}"
+      meta_results = meta_response.json()
+
+      if not meta_results:
+        return response_error_handler({"status": 404})
+
+      if app == "entity-viewer":
+        url = f"{ENSEMBL_URL}/{app}/{genome_id}/{type}:{stable_id}"
+      else:
+        url = f"{ENSEMBL_URL}/{app}/{genome_id}?focus={type}:{stable_id}"
+
+      meta = {
+        "accession_id": meta_results["assembly"]["accession_id"],
+        "assembly_name": meta_results["assembly"]["name"],
+        "species_scientific_name": meta_results["scientific_name"],
+        "taxonomy_id": meta_results["taxonomy_id"],
+        "resolved_url": url
+      }
+
+      resolved_genome_info.append(meta)
+    
+  if request.headers.get("content-type") == "application/json":
+    return resolved_genome_info
```

Comment:
`resolved_genome_info` is list here and not JSON if that is what is expected here. May be FastAPI could be converting it to JSON but did you check if it is the case ?

Reply:
Fastapi returns a Json.